### PR TITLE
Create new routing tutorial that describes how to refactor routes.

### PR DESCRIPTION
**Pull request by aikithoughts**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Tutorial on refactoring routes does not exist.

Issue Number: N/A

## What is the new behavior?
New tutorial on refactoring routes in an Angular application.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: aio/content/examples/router-refactor-tutorial/src/app/app.component.css [hunk 1]
```diff
@@ -0,0 +1,34 @@
+.button {
```
Comment (kapunahelewong) [inline aio/content/examples/router-refactor-tutorial/src/app/app.component.css:1]: Same comment here for using button as a class name. Also applies to `activeButton`.

File: aio/content/examples/router-refactor-tutorial/src/app/app.module.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { BrowserModule } from '@angular/platform-browser';
+import { NgModule } from '@angular/core';
+import { RouterModule } from '@angular/router';
+import { AppComponent } from './app.component';
+import { CrisisListComponent } from './crisis-list/crisis-list.component';
+import { HeroesListComponent } from './heroes-list/heroes-list.component';
+import { PageNotFoundComponent } from './page-not-found/page-not-found.component';
+
+// #docplaster
```
Comment (kapunahelewong) [inline aio/content/examples/router-refactor-tutorial/src/app/app.module.ts:9]: `// #docplaster` should go on line one.

File: aio/content/guide/router-refactoring-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+You can view or download the code from this tutorial: <live-example name="router-tutorial"></live-example>.
+
+## Move routes into a separate array
+
+Currently, your application defines routes inside the `imports` array in the `app.module.ts` file.
+
+<code-example header="src/app/app.module.ts" path="router-tutorial/src/app/app.module.ts" region="import-wildcard"></code-example>
+
+This `imports` array can contain a lot of information about your application.
+While a basic application might contain only a couple of routes, more complicated applications could contain much more.
```
Comment (kapunahelewong) [inline aio/content/guide/router-refactoring-tutorial.md:34]: Many more routes or much more complexity/configuration?

File: aio/content/guide/router-refactoring-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+1. Copy your existing routes to the new array.
+
+   <code-example header="src/app/app.module.ts" path="router-refactor-tutorial/src/app/app.module.ts" region="routes-array"></code-example>
+
+1. Update the `imports` array in the `@NgModule` statement.
+
+   <code-example header="src/app/app.module.ts" path="router-refactor-tutorial/src/app/app.module.ts" region="imports"></code-example>
+
+View your application in a browser window. You should see that your application runs as expected, and all of your routes work without errors.
```
Comment (kapunahelewong) [inline aio/content/guide/router-refactoring-tutorial.md:49]:
```suggestion
View your application in a browser window. 
You should see that your application runs as expected, and all of your routes work without errors.
```

Hard carriage return for new sentence.

File: aio/content/guide/router-refactoring-tutorial.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+1. Open a terminal window, and navigate to your project directory: `angular-router-sample`.
+
+1. Run the `ng generate module` command with the following parameters.
+
+   <code-example lang="sh">
+   ng generate module app-routing --flat --module=app
+   </code-example>
+
+   This command creates a new module, `app-routing`. The additional flags perform these tasks:
```
Comment (kapunahelewong) [inline aio/content/guide/router-refactoring-tutorial.md:67]:
```suggestion
   This command creates a new module, `app-routing`. 
   The additional flags perform these tasks:
```
Hard carriage return for new sentence.

Note: Within a list you just need to make sure that the indentation is maintained on new lines.

File: aio/content/guide/router-refactoring-tutorial.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+1. In the `imports` array of the `@NgModule` statement, replace `RouterModule.forRoot(appRoutes)` with `AppRoutingModule`.
+
+   <code-example header="src/app/app.module.ts" path="router-refactor-tutorial-with-module/src/app/app.module.ts" region="imports"></code-example>
+
+1. Delete `RouterModule` from your list of import statements at the top of the file. Your new `app.module.ts` file should appear as follows:
+
+   <code-example header="src/app/app.module.ts" path="router-refactor-tutorial-with-module/src/app/app.module.ts"></code-example>
+
+View your application in a browser window. Again, you should see that your application's behavior remains unchanged. However, you now have configured your application's routes in a way that is easy to maintain as your application grows in scope and complexity.
```
Comment (kapunahelewong) [inline aio/content/guide/router-refactoring-tutorial.md:123]:
```suggestion
View your application in a browser window. Again, you should see that your application's behavior remains unchanged. 
However, you now have configured your application's routes in a way that is easy to maintain as your application grows in scope and complexity.
```

Hard carriage return for new sentence.

Comment (kapunahelewong) [inline aio/content/guide/router-refactoring-tutorial.md:123]:
```suggestion
View your application in a browser window. Again, you should see that your application's behavior remains unchanged. 
However, you now have configured your application's routes in a way that is easy to maintain as your application grows in scope and complexity.
```

File: aio/content/navigation.json [hunk 7]
```diff
@@ -754,6 +754,11 @@
           "url": "guide/router-tutorial",
           "title": "Using Angular Routes in a Single-page Application",
           "tooltip": "A tutorial that covers many patterns associated with Angular routing."
+        },
+        {
+          "url": "guide/router-refactoring-tutorial",
+          "title": "Refactoring routes in Angular",
```
Comment (kapunahelewong) [inline aio/content/navigation.json:760]: Nice title!

**Review by kapunahelewong — APPROVED** [review]
Love that refactoring is a separate tutorial!

**Comment by jessicajaniuk** [comment]
@aikidave This PR seems to have stalled. Any chance we could get an update on whether you still want to land it?

**Comment by aikidave** [comment]
Hi @jessicajaniuk ! Thanks for bringing this up. I will definitely have a look. It may take me a week or so to fully review it, however.

**Pull request closed without merging**